auto_nag/scripts/regression_without_regressed_by.py

```diff
@@ -0,0 +1,207 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import dateutil.parser
+from libmozdata.bugzilla import Bugzilla, BugzillaUser
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag import utils
+
+
+class RegressionWithoutRegressedBy(BzCleaner):
+    def __init__(self):
+        super(RegressionWithoutRegressedBy, self).__init__()
+
+    def description(self):
+        return 'Regressions without regressed_by and some dependencies'
+
+    def handle_bug(self, bug, data):
+        bugid = bug['id']
+        deps = set(bug['blocks']) | set(bug['depends_on'])
+        assignee = bug['assigned_to_detail']
+        if utils.is_no_assignee(assignee['email']):
+            assignee = None
+
+        data[str(bugid)] = {
+            'deps': deps,
+            'assignee': assignee,
+            'creator': bug['creator_detail'],
+            'creation': dateutil.parser.parse(bug['creation_time']),
+        }
+        return bug
+
+    def filter_bugs(self, bugs):
+        all_deps = set()
+        dep_bug_creation = {}
+
+        for bugid, info in bugs.items():
+            bugid = int(bugid)
+            info['deps'] = deps = set(x for x in info['deps'] if x < bugid)
+            if deps:
+                all_deps |= deps
+                for dep in deps:
+                    dep_bug_creation[dep] = info['creation']
+
+        def bug_handler(bug, data):
+            if 'meta' in bug['keywords'] or not bug['cf_last_resolved']:
+                data.add(bug['id'])
+
+        def history_handler(bug, data):
+            bugid = bug['id']
+            resolved_before = False
+            for h in bug['history']:
+                if resolved_before:
+                    break
+                for change in h['changes']:
+                    if change.get(
+                        'field_name', ''
+                    ) == 'cf_last_resolved' and change.get('added', ''):
+                        date = dateutil.parser.parse(change['added'])
+                        if date < dep_bug_creation[bugid]:
+                            resolved_before = True
+                            break
+            if not resolved_before:
+                data.add(bugid)
+
+        invalids = set()
+        Bugzilla(
+            bugids=list(all_deps),
+            include_fields=['id', 'keywords', 'cf_last_resolved'],
+            bughandler=bug_handler,
+            bugdata=invalids,
+            historyhandler=history_handler,
+            historydata=invalids,
+        ).get_data().wait()
+
+        to_rm = []
```

Comment:
Nit: It'd be nicer to have a comprehension rather than storing a list of bugs to remove and then removing them